### PR TITLE
[#522] Update to Hono 2.5.0

### LINKED ISSUE #522: Update to Hono 2.5.0
Update Hono chart to use Hono 2.5.0. In doing so, we should also change the Apache Kafka broker configuration to use KRaft mode (i.e. without running Apache Zookeeper).

### PR DESCRIPTION
**Pull request by sophokles73**
Changed configuration to run Apache Kafka in KRaft mode.

Fixes #522

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn Would you mind taking a look?

File: charts/hono/Chart.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@ name: hono
 description: |
   Eclipse Hono™ provides remote service interfaces for connecting large numbers of IoT devices to a back end and
   interacting with them in a uniform way regardless of the device communication protocol.
-version: 2.5.6
+version: 2.5.7
```
Comment (calohmn) [inline charts/hono/Chart.yaml:18]: I think the changes in this PR would rather require a `2.6.0` version here.

File: charts/hono/values.yaml [hunk 2]
```diff
@@ -1724,82 +1724,68 @@ kafka:
...
-  replicaCount: 1
   # Do not write data to a PersistentVolume by default.
   # When setting this to true, make sure to also set "kafka.zookeeper.persistence.enabled" to true and
   # make sure that Kafka is stopped cleanly using SIGTERM _before_ Zookeeper is stopped.
   # Otherwise you will most likely run into issues like
   # https://stackoverflow.com/questions/39759071/error-while-starting-kafka-broker
-  persistence:
-    enabled: false
+#  persistence:
+#    enabled: false
```
Comment (calohmn) [inline charts/hono/values.yaml:1765]: It looks like there is no top-level `persistence` property anymore in the Kafka-chart. 
Also the mentioning of the `kafka.zookeeper.persistence.enabled` setting above could be rather confusing, now that KRaft mode is used by default.

**Comment by calohmn** [comment]
I think it would be good to also add support for the new `useLegacyTraceContextFormat` property in the new chart version.
I've created #524 for that (to be merged in this PR branch).

**Comment by sophokles73** [comment]
sounds good, I will do that 👍

**Comment by sophokles73** [comment]
@calohmn would you mind taking another look?

**Comment by calohmn** [comment]
The Kafka chart update allows us to update the MongoDB chart again - see #490.
I think we could at least use the MongoDB chart version `^13.x` again (undoing the change from #491).

**Comment by sophokles73** [comment]
@calohmn I have changed the MongoDB dependency ...

**Comment by sophokles73** [comment]
@calohmn I will now squash this into two commits and then we're good to go, right?

**Review by calohmn — APPROVED** [review]
@sophokles73 yes, LGTM.

**Pull request merged**